erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -32,6 +32,36 @@ def onload(self):
 
 	def on_update(self):
 		self.update_nsm_model()
+		self.create_warehouse_address()
+		self.create_warehouse_contact()
+
+	def create_warehouse_address(self):
```

Comment:
docstrings and comments please. Every function should have a docstring and every block inside it should have a header comment of sorts